### PR TITLE
Remove MICROPY_PORT_* macros used for never reset

### LINKED ISSUE #4997: Auto-Reload not releasing APA102_MOSI on Trinket_M0
### CircuitPython version

```python
Adafruit CircuitPython 7.0.0-alpha.4-49-g9fdecacec on 2021-07-13; Adafruit Trinket M0 with samd21e18
```

### Code/REPL

```python
code.py:

import adafruit_dotstar
import board

pixel = adafruit_dotstar.DotStar(board.DOTSTAR_CLOCK, board.DOTSTAR_DATA, 1)

while True:
    pixel.fill((255, 255, 0))
```

### Behavior

Runs normally and continues to run normally after a  Control-D soft reboot

but if I write soemthing to the CIRCUITPY drive whil it it running, it does an "auto-reload" and fails
on Linux I did `touch /media/jerryneedell/CIRCUITPY/junk` while it was running

caused:
```
Auto-reload is on. Simply save files over USB to run them or enter REPL to disable.
code.py output:
Traceback (most recent call last):
  File "code.py", line 4, in <module>
  File "adafruit_dotstar.py", line 112, in __init__
ValueError: APA102_MOSI in use

Code done running.
```

### Description

_No response_

### Additional information

_No response_

### LINKED ISSUE #5001: Onboard NeoPixel still in use after filesystem change triggered restart.
### CircuitPython version

```python
Adafruit CircuitPython 7.0.0-alpha.4 on 2021-07-08; Adafruit Metro M4 Express with samd51j19
```

### Code/REPL

```python
import time
import board
from neopixel import NeoPixel

pixel = NeoPixel(board.NEOPIXEL, 1, brightness=0.3)

lightshow = [0xFF0000, 0xCCCC00, 0x00CC00, 0xCCCC00]
step = 0
while True:
    pixel[0] = lightshow[step % len(lightshow)]
    time.sleep(0.1)
    step += 1
```

### Behavior

```python
Traceback (most recent call last):
  File "code.py", line 14, in <module>
  File "neopixel.py", line 127, in __init__
ValueError: NEOPIXEL in use
```
When a filesystem change triggers a reset the above error appears.

### Description

Using Ctrl-C Ctrl-D does work. The NEOPIXEL pin is not in use when code.py starts.

It is annoying to have to reset the board after saving a code change.

I may resort to this:

```python
try:
    pixel = NeoPixel(board.NEOPIXEL, 1, brightness=0.3)
except ValueError:
    print("Sorry, neopixel didn't get reset.\nHang on...")
    import microcontroller
    microcontroller.reset()
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by tannewt**
Now we use never_reset which is generally more correct and easier
to get right.

Fixes #5001 and fixes #4997

### CODE REVIEW

**Comment by tannewt** [comment]
@itacainnovation please review this for your board. I wasn't sure why you had some pins set to not reset.

File: ports/atmel-samd/boards/trinket_m0_haxpress/mpconfigboard.h [hunk 1]
```diff
@@ -11,11 +11,6 @@
 #define SPI_FLASH_SCK_PIN           &pin_PA17
 #define SPI_FLASH_CS_PIN            &pin_PA11
 
-// These are pins not to reset.
-#define MICROPY_PORT_A        (PORT_PA00 | PORT_PA01 | PORT_PA18)
```
Comment (dhalbert) [inline ports/atmel-samd/boards/trinket_m0_haxpress/mpconfigboard.h:15]: PA00 and PA01 are the APA102 pins, but they are commented out on lines 6 and 7. Is that a mistake? They are not commented out in the regular `trinket_m0/mpconfigboard.h`

**Review by dhalbert — COMMENTED** [review]
This is a nice cleanup! Just one question.

File: ports/atmel-samd/boards/trinket_m0_haxpress/mpconfigboard.h (hunk 1 above)
Comment (tannewt) [reply ports/atmel-samd/boards/trinket_m0_haxpress/mpconfigboard.h:15]: I don't know. I assumed they may share flash pins.

Comment (dhalbert) [reply ports/atmel-samd/boards/trinket_m0_haxpress/mpconfigboard.h:15]: I looked at the schematic, and PA00 and PA01 are still just connected to the NeoPixel. So I think those lines should be uncommented.
image

**Comment by dhalbert** [comment]
The alorium tech board needs a `#include`, and it looks like the cache key for xtensa needs to be bumped again: all those builds failed with the typical error caused by a bad cache :/

**Review by dhalbert — APPROVED** [review]
:+1:

**Comment by ITACAInnovation** [comment]
> @ITACAInnovation please review this for your board. I wasn't sure why you had some pins set to not reset.

Hi @tannewt 
thanks for the help, the pins should never reset because they control some hardware which should keep pre-defined values in order not to cause specific situations issues. I will check the compiled source on a board and be back to you with eventual changes to the source.

**Pull request merged**